src/model/log_prob.jl

```diff
@@ -395,44 +391,40 @@ function init_galaxy_state(entry::CatalogEntry)
 end
 
 
-function elbo_args_vp_to_star_state(vp::Array{Float64,1})
-    fluxes      = variational_params_to_fluxes(vp)
-    lnr, colors = fluxes_to_colors(fluxes)
-    ra, dec     = vp[ids.u[1]], vp[ids.u[2]]
-    return [[lnr]; colors; [ra, dec]]
-end
+function catalog_entry_to_latent_state_params(ce::CatalogEntry)
+    # create a float array of the appropriate length
+    ret = Array(Float64, length(lidx))
 
+    # galaxy shape params
+    ret[lidx.u]       = ce.pos
+    ret[lidx.e_dev]   = ce.gal_frac_dev
+    ret[lidx.e_axis]  = ce.gal_ab
+    ret[lidx.e_scale] = ce.gal_scale
 
-function elbo_args_vp_to_galaxy_state(vp::Array{Float64, 1})
-    shared_params = elbo_args_vp_to_star_state(vp)
-    gal_shape = [vp[ids.e_dev]  ,
-                 vp[ids.e_axis] ,
-                 vp[ids.e_angle],
-                 vp[ids.e_scale]]
-    return [shared_params; unconstrain_gal_shape(gal_shape)]
+    # star, gal r flux
+    star_lnr, star_cols = fluxes_to_colors(ce.star_fluxes)
+    gal_lnr, gal_cols   = fluxes_to_colors(ce.gal_fluxes)
+    ret[lidx.r]         = [star_lnr, gal_lnr]
+    ret[lidx.c]         = hcat([star_cols, gal_cols]...)
+
+    # set the prob star/prob gal
+    ret[lidx.a] = clamp(ce.is_star, EPS, 1-EPS)
+    ret
 end
 
 
-function variational_params_to_fluxes(i::Int, vs::Array{Array{Float64,1}})
-    ret = Array(Float64, 5)
-    ret[3] = exp(vs[ids.r1[i]] + 0.5 * vs[ids.r2[i]])
-    ret[4] = ret[3] * exp(vs[ids.c1[3, i]])
-    ret[5] = ret[4] * exp(vs[ids.c1[4, i]])
-    ret[2] = ret[3] / exp(vs[ids.c1[2, i]])
-    ret[1] = ret[2] / exp(vs[ids.c1[1, i]])
-    ret
+function extract_star_state(ls::Array{Float64, 1})
+    return [[ls[lidx.r[1]]]; ls[lidx.c[:, 1]]; ls[lidx.u]]
```

Comment:
Thanks!  

One issue with substituting tuples here is that MCMC and MAP estimation will be doing vector/matrix computations with the parameter vector.  This conversion won't be called at each iteration --- just to start, so it might be premature to rewrite code to optimize this before I benchmark mcmc routines.